vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/TouchActions.java

```diff
@@ -224,10 +224,24 @@ private Supplier<BufferedImage> createScreenShooter(SearchContext searchContext)
      */
     public void performVerticalSwipe(int startY, int endY, Rectangle swipeArea, Duration swipeDuration)
     {
-        swipe(SwipeDirection.createCoordinates(startY, endY, swipeArea.getWidth(),
+        swipe(SwipeDirection.createYCoordinates(startY, endY, swipeArea.getWidth(),
                 mobileApplicationConfiguration.getSwipeVerticalXPosition(), swipeArea.getPoint()), swipeDuration);
     }
 
+    /**
+     * Performs horizontal swipe from <b>startX</b> to <b>endX</b> with <b>swipeDuration</b>
+     *
+     * @param startX        start X coordinate
+     * @param endX          end X coordinate
+     * @param swipeArea     the area to execute the swipe
+     * @param swipeDuration swipe duration in <a href="https://en.wikipedia.org/wiki/ISO_8601">ISO 8601</a> format
+     */
+    public void performHorizontalSwipe(int startX, int endX, Rectangle swipeArea, Duration swipeDuration)
+    {
+        swipe(SwipeDirection.createXCoordinates(startX, endX, swipeArea.getHeight(),
+                mobileApplicationConfiguration.getSwipeHorizontalYPosition(), swipeArea.getPoint()), swipeDuration);
+    }
```

Comment:
why not to have a single method, that will accept swipe direction as an input parameter?

Reply:
done